src/Commands/NotificationQueuer.php

```diff
@@ -92,6 +100,57 @@ public function listen($media_type_id) {
     $this->io()->progressFinish();
   }
 
+  /**
+   * Removes notifications that are older than the entities that reference them.
+   *
+   * @param \Lullabot\Mpx\DataService\Notification[] $notifications
+   *   An array of notifications.
+   * @param string $media_type_id
+   *   The media type being imported.
+   *
+   * @return array
+   *   The filtered array.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   *   The only way this could happen is if the media module was missing
+   *   (thrown from getStorage call).
+   */
+  private function filterByDate(array $notifications, string $media_type_id): array {
+    /* @todo I borrowed this search code from the DataObjectImporter
+     * maybe there is a better place to publicly store this functionality. */
+
+    /** @var \Drupal\Media\MediaStorage $media_storage */
+    $media_storage = \Drupal::entityManager()->getStorage('media');
+    /** @var \Drupal\Media\Entity\MediaType $media_type */
+    $media_type = MediaType::load($media_type_id);
+    $source = $media_type->getSource();
+    $source_field = $source->getSourceFieldDefinition($media_type)->getName();
+
+    $notifications = array_filter($notifications, function (MpxNotification $notification) use ($media_storage, $source_field) {
+      $notificationDate = $notification->getEntry()->getUpdated()->format("U");
+      $notificationId = (string) $notification->getEntry()->getId();
+      $entities = $media_storage->loadByProperties([$source_field => $notificationId]);
+
+      // The updates must have to do with something new so keep them included.
+      if (empty($entities)) {
+        return TRUE;
+      }
+
+      /* If there exists an entity that hasn't been updated since the
```

Comment:
I'm surprised this passes phpcs, but normally the convention is `//` for inline coments. But if it passes, it passes :D